app/src/main/java/com/glia/exampleapp/MainFragment.kt

```diff
@@ -579,7 +581,19 @@ class MainFragment : Fragment() {
 
     private fun deAuthenticate() {
         if (activity == null || containerView == null) return
-        authentication!!.deauthenticate( {
+        MaterialAlertDialogBuilder(requireContext())
+            .setTitle("Stop Push Notifications")
+            .setPositiveButton("Yes", DialogInterface.OnClickListener { _, _ ->
+                deAuthenticateInternal(true)
+            })
+            .setNegativeButton("No", DialogInterface.OnClickListener { _, _ ->
+                deAuthenticateInternal(false)
+            }).show()
+    }
+
+    private fun deAuthenticateInternal(stopPushNotifications: Boolean) {
+        if (activity == null || containerView == null) return
+        authentication!!.deauthenticate(stopPushNotifications, {
```

Comment:
NIT. Might be changed to `authentication?.deauthenticate()`.

Reply:
fixed